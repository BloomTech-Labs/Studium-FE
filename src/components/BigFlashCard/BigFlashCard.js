import React, {useState} from 'react';
import styled from 'styled-components';
import {Card} from 'antd';
import PropTypes from 'prop-types';

/**
 * Big Flash Card
 *
 * @component
 * @example
 * const flashCard = {question: "Question?", answer: "Answer"}
 * return (
 *  <BigFlashCard flashCard={flashCard} />
 *  )
 */
const BigFlashCard = ({flashCard}) => {
  const [position, setPosition] = useState('front');
  
  const flipCard = () => {
    
    const newPos = position === 'front' ? 'back' : 'front';
    setPosition(newPos);
  };
  
  return (
    <StyledCardContainer data-testid="StyledCardContainer"
                         style={{position: 'relative'}} onClick={flipCard}>
      <StyledCard
        position={position}
        
        style={{width: '285px', height: '421.56px'}}
      >
        <CardText>
          {position === 'front' ? flashCard.question : flashCard.answer}
        </CardText>
      </StyledCard>
    </StyledCardContainer>
  
  );
};

BigFlashCard.prototypes = {
  flashCard: PropTypes.objectOf({
    question: PropTypes.string.isRequired, answer: PropTypes.string.isRequired,
  }).isRequired,
};

const StyledCardContainer = styled.div`
  height: 390px;
  width: 283px;
  position: relative;
`;

const StyledCard = styled(Card)`
  && {
    margin: 12px auto 0 auto;
    background: ${props => props.position === 'front' ? '#F7F7F7' :
  '#1b1414c9'};
    color: ${props => (props.position === 'front' ? '#1b1414c9' : 'white')};

    border-radius: 11px;
    > .ant-card-body {
      min-height: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }
  transition: all 1s;
`;
const CardText = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  font-size: 20px;
  font-weight: 900;
  transition: all 1s;
`;
export default BigFlashCard;
