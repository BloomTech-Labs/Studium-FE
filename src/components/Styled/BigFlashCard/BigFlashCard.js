import React, { useState } from 'react';
import styled from 'styled-components';
import card2 from '../../../images/Card2.jpg';
import card3 from '../../../images/Card3.jpg';
import { Card } from 'antd';
import PropTypes from 'prop-types';

/**
 * Big Flash Card
 *
 * @component
 * @example
 * const flashCard = {question: "Question?", answer: "Answer"}
 * return (
 *  <BigFlashCard flashCard={flashCard} />
 *  )
 */
const BigFlashCard = ( { flashCard } ) => {
  const [ position, setPosition ] = useState( 'front' );
  
  const flipCard = () => {
    
    const newPos = position === 'front' ? 'back' : 'front';
    setPosition( newPos );
  };
  
  return (
    <StyledCardContainer data-testid = "StyledCardContainer" style={{ position: 'relative' }} onClick={flipCard}>
      <StyledCard
        position={position}
      
        style={{ width: '285px', height: '421.56px' }}
      >
        <CardText>
          { position === 'front' ? flashCard.question : flashCard.answer }
        </CardText>
      </StyledCard>
      <img
        src={ card2 }
        style={ {
          position: 'absolute',
          top: '14px',
          height: '417px',
          width: '276px',
          zIndex: -1,
          left: '5px',
        } }
        alt={ 'card' }
      />
      <img
        src={ card3 }
        style={ {
          position: 'absolute',
          top: '22px',
          left: '7px',
          height: '417px',
          width: '260px',
          zIndex: -3,
        } }
        alt={ 'card' }
      />
    </StyledCardContainer>
  
  );
};

BigFlashCard.prototypes = {
  flashCard: PropTypes.objectOf( {
    question: PropTypes.string.isRequired, answer: PropTypes.string.isRequired,
  } ).isRequired,
};

const StyledCardContainer = styled.div`
  height: 390px;
  width: 283px;
  position: relative;
`;

const StyledCard = styled( Card )`
  && {
    margin: 121px auto 0 auto;
    background: ${ props => props.position === 'front' ? '#F7F7F7' :
  '#1b1414c9' };
    color: ${ props => ( props.position === 'front' ? '#1b1414c9' : 'white' ) };

    border-radius: 11px;
    > .ant-card-body {
      min-height: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }
  transition: all 1s;
`;
const CardText = styled.div`
  height: 100%;
  display: flex;
  justifycontent: center;
  font-size: 20px;
  font-weight: 900;
  transition: all 1s;
`;
export default BigFlashCard;
