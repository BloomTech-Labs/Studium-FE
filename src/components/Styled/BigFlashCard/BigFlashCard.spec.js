import React from 'react';
import {
  customRender,
  getNodesByType,
  getByTestId,
  fireEvent,
  store,
  getByRole,
  logOutMessageOrDebug,
  getByText,
  toHaveTextContent,wait
} from '../../../utilities/test-utils';
import BigFlashCard from './BigFlashCard';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';



/**
 * Test the big flash card
 */

// test('BigFlashCard render correctly', () => {
//   const tree = renderer.create(<BigFlashCard flashCard={{question: "This is" +
//       " a question", answer: "This is the answer."}} />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

describe('Big Flash Card', () => {

  test('BigFlashCard render correctly', () => {
    const tree = renderer.create(<BigFlashCard flashCard={{question: "This is" +
        " a question"}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

    test('position front', () => {
    const { container, getByText } = render(<BigFlashCard flashCard = {{question: "question", answer: "answer"}}/>)
    expect(getByText('question')).toBeInTheDocument()
  })
    test('position back', () => {
    const { container, getByTestId, queryByText, debug} = render(<BigFlashCard flashCard = {{question: "question", answer: "answer"}}/>)
    const card = getByTestId('StyledCardContainer')
    fireEvent.click(card)
    expect(queryByText('answer')).toBeInTheDocument()
    //debug(queryByText('answer'))
  })

 
});