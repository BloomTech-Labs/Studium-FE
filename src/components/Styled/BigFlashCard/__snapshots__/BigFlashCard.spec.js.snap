// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Big Flash Card BigFlashCard render correctly 1`] = `
.c0 {
  height: 390px;
  width: 283px;
  position: relative;
}

.c1 {
  -webkit-transition: all 1s;
  transition: all 1s;
}

.c1.c1 {
  margin: 121px auto 0 auto;
  background: #F7F7F7;
  color: #1b1414c9;
  border-radius: 11px;
}

.c1.c1 > .ant-card-body {
  min-height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  justifycontent: center;
  font-size: 20px;
  font-weight: 900;
  -webkit-transition: all 1s;
  transition: all 1s;
}

<div
  className="c0"
  data-testid="StyledCardContainer"
  onClick={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="ant-card c1 ant-card-bordered"
    position="front"
    style={
      Object {
        "height": "421.56px",
        "width": "285px",
      }
    }
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="c2"
      >
        This is a question
      </div>
    </div>
  </div>
  <img
    alt="card"
    src="Card2.jpg"
    style={
      Object {
        "height": "417px",
        "left": "5px",
        "position": "absolute",
        "top": "14px",
        "width": "276px",
        "zIndex": -1,
      }
    }
  />
  <img
    alt="card"
    src="Card3.jpg"
    style={
      Object {
        "height": "417px",
        "left": "7px",
        "position": "absolute",
        "top": "22px",
        "width": "260px",
        "zIndex": -3,
      }
    }
  />
</div>
`;
